### PR TITLE
runtime-builder: replace debian8 image with distroless base image for the runtime image

### PR DESCRIPTION
**Pull request by shantuo**
Keep debian_digest used by the builder as a LABEL in the runtime image for reproducible build.

remove go-run.sh since it won't work with distroless image.

### CODE REVIEW

File: runtime-builder/build.sh [hunk 1]
```diff
@@ -26,7 +26,7 @@ usage() { echo "Usage: $0 <project> <go_version>"; exit 1; }
 debian_digest()
 {
   # We sometimes have issue with 'describe', use the --log-http flag to get more info if it fails.
-  local digest="$(gcloud container images describe gcr.io/google-appengine/debian8:latest | \
+  local digest="$(gcloud alpha container images describe gcr.io/google-appengine/debian8:latest | \
```
Comment (zombiezen) [inline runtime-builder/build.sh]: Shouldn't this be the digest for distroless/base?

Comment (shantuo) [inline runtime-builder/build.sh]: No, this is to record which version we use for the builder.

We can also record distroless/base digest, that will be in go-build.sh, when building the output image.

Comment (cybrcodr) [inline runtime-builder/build.sh]: I don't think there is a need to record the debian digest in the user's image.  

(1) It's implementation detail of the builder and that may change.
(2) A debian image can be used by different versions of the builder.

What perhaps you may want to record in the user's image is the builder's unique tag, i.e. use BUILD_TAG env below.

File: runtime-builder/go-build.sh [hunk 2]
```diff
@@ -44,8 +44,9 @@ mv "${staging}" "${workspace}"/app
 
 # Generate application Dockerfile.
 cat > Dockerfile <<EOF
-FROM gcr.io/google-appengine/debian8@${DEBIAN_DIGEST}
+FROM gcr.io/distroless/base:latest
```
Comment (cybrcodr) [inline runtime-builder/go-build.sh]: Don't use "latest".  Use the same technique that we did for grabbing debian8's latest digest tag.

File: runtime-builder/tests/integration/go-test.yaml [hunk 3]
```diff
@@ -1,7 +1,6 @@
 steps:
 - name: gcr.io/gcp-runtimes/integration_test
   args: [
-    '--no-deploy',
```
Comment (cybrcodr) [inline runtime-builder/tests/integration/go-test.yaml]: Curious.  What does --no-deploy do?

File: runtime-builder/build.sh (hunk 1 above)
Comment (dlorenc) [inline runtime-builder/build.sh]: Do you need to use alpha here?

Comment (shantuo) [inline runtime-builder/build.sh]: > What perhaps you may want to record in the user's image is the builder's unique tag, i.e. use BUILD_TAG env below.
Use BUILD_TAG sounds fine. Will change this to recording distroless digest.

> Do you need to use alpha here?
It got demoted to alpha in the most recent release 166.0.0

File: runtime-builder/go-build.sh (hunk 2 above)
Comment (shantuo) [inline runtime-builder/go-build.sh]: Ack.

File: runtime-builder/tests/integration/go-test.yaml (hunk 3 above)
Comment (nkubala) [inline runtime-builder/tests/integration/go-test.yaml]: `--no-deploy` used to explicitly tell the integration testing framework not to deploy the sample app. now, it's implied if you pass the `--url` flag

**Pull request merged**